src/pybind/mgr/cephadm/services/nvmeof.py

```diff
@@ -89,6 +89,22 @@ def get_set_cmd_dicts(out: str) -> List[dict]:
                     logger.warning(f'No ServiceSpec found for {dd.service_name()}')
                     continue
 
+                self.mgr.check_mon_command({
+                    'prefix': 'config-key set',
+                    'key': 'mgr/dashboard/nvmeof/mtls_server_cert',
+                    'val': spec.server_cert if spec.server_cert else '',
+                })
+                self.mgr.check_mon_command({
+                    'prefix': 'config-key set',
+                    'key': 'mgr/dashboard/nvmeof/mtls_client_key',
+                    'val': spec.server_key if spec.server_key else '',
+                })
+                self.mgr.check_mon_command({
+                    'prefix': 'config-key set',
+                    'key': 'mgr/dashboard/nvmeof/mtls_client_cert',
+                    'val': spec.client_cert if spec.client_cert else '',
```

Comment:
why is it we don't need the client key? Just asking in case we're expecting this client cert to be in PEM format and that's why we're not grabbing a separate key because currently cephadm is expecting separate `client_cert` and `client_key` fields to be set in the nvmeof service spec.

Reply:
@adk3798 what's missing is the server_key. I copied the implementation from [here:](https://github.com/ceph/ceph-nvmeof/blob/fb843d9dc11f4e66a7bd3471ad8c7f22df3f6d11/control/cli.py#L227). The CLI only pass server_cert, client_key and client_cert. It doesn't pass the server_key anywhere. I copied the same implementation here.

From testing perspective, I only verified the config is correctly set or not. I need to verify the mTLS part. I'll try to verify that and let you know if this works or not.

Reply:
The certificate chain of trust is or should be achieved through shared trust in a Root Certificate Authority (a parent certificate whose private key signed both client and server certs).

As these are self-signed certificates, either the client or the server certificate behaves as the root CA (there's no chain of trust, but stand-alone certs), but ideally it should be the same Root CA cert for both.

So the input params should be:
* For the server: server key, server cert, and root CA cert (instead of client cert).
* For the client: client key, client cert, and rooc CA cert (instead of server cert).